aws-ecs/60-ecs.tf

```diff
@@ -32,59 +32,59 @@ resource "aws_security_group" "service" {
 resource "aws_ecs_task_definition" "pvault" {
   family = "${var.deployment_id}-task"
 
-  container_definitions = <<EOF
-  [
-    {
-      "name": "${var.deployment_id}-container",
-      "image": "${var.pvault_repository}:${var.pvault_tag}",
-      "entryPoint": [],
-      "environment": [
-        {"name": "PVAULT_DEVMODE", "value": "1"},
-        {"name": "PVAULT_DB_HOSTNAME", "value": "${module.db.db_instance_address}"},
-        {"name": "PVAULT_DB_NAME", "value": "${var.rds_db_name}"},
-        {"name": "PVAULT_DB_USER", "value": "${var.rds_username}"},
-        {"name": "PVAULT_DB_PORT", "value": "${var.rds_port}"},
-        {"name": "PVAULT_LOG_CUSTOMER_IDENTIFIER", "value": "${var.pvault_log_customer_identifier}"},
-        {"name": "PVAULT_LOG_CUSTOMER_ENV", "value": "${var.pvault_log_customer_env}"},
-        {"name": "PVAULT_KMS_URI", "value": "aws-kms://${aws_kms_key.pvault.arn}"}
-      ],
-      "secrets": [
-        {"name": "PVAULT_DB_PASSWORD", "valueFrom": "${aws_secretsmanager_secret.db_password.arn}"},
-        {"name": "PVAULT_SERVICE_LICENSE", "valueFrom": "${aws_secretsmanager_secret.pvault_service_license.arn}"},
-        {"name": "PVAULT_SERVICE_ADMIN_API_KEY", "valueFrom": "${aws_secretsmanager_secret.pvault_service_admin_api_key.arn}"}
-      ],
-      "essential": true,
-      "logConfiguration": {
-        "logDriver": "awslogs",
-        "options": {
-          "awslogs-group": "${aws_cloudwatch_log_group.pvault.name}",
-          "awslogs-region": "${var.aws_region}",
-          "awslogs-stream-prefix": "pvault"
-        }
-      },
-      "portMappings": [
-        {
-          "containerPort": 8123,
-          "hostPort": 8123,
-          "protocol": "tcp",
-          "name": "api"
-        }
-      ],
-      "healthCheck": {
-          "command": [
-              "CMD-SHELL",
-	        "wget --spider localhost:8123/api/pvlt/1.0/ctl/info/health"
+  container_definitions = jsonencode(
+    [
+      {
+        name       = "${var.deployment_id}-container",
+        image      = "${var.pvault_repository}:${var.pvault_tag}",
+        entryPoint = [],
+        environment = concat([for k, v in var.pvault_env_vars : { "name" = k, "value" = v }], [
```

Comment:
why you don't use same as in apprunner (with merge) ?

Reply:
Because apprunner accepts a map, and apperantly ecs accepts a list of `[{"name": $name, "value": $value}, ...]`. The function `merge` merges maps, while `concat` works on lists

Reply:
But I can improve it. Let me push another commit